clippy_lints/src/unused_io_amount.rs

```diff
@@ -47,58 +47,54 @@ impl<'tcx> LateLintPass<'tcx> for UnusedIoAmount {
                         func.kind,
                         hir::ExprKind::Path(hir::QPath::LangItem(hir::LangItem::TryIntoResult, _))
                     ) {
-                        check_method_call(cx, &args[0], expr);
+                        match args[0].kind {
+                            hir::ExprKind::MethodCall(ref path, _, ref args, _) => {
+                                if matches!(&*path.ident.as_str(), "or" | "or_else" | "ok") {
```

Comment:
This will not cover `or().expect()`. I think a loop would solve this well.

It might be better to have a single `check_method_call` invocation at the bottom with early returns.

Reply:
You can break it into three steps (read the comments from bottom to top)

```
read()       // finally, get this
  .or().or() // second, traverse these
  .expect() // first get this, or `?`
```